src/hooks/useSupply.ts

```diff
@@ -0,0 +1,30 @@
+import { Asset, VTokenId } from 'types';
+import {
+  useSupply as useSupplyNonBnb,
+  useSupplyBnb,
+  ISupplyInput,
+  ISupplyBnbInput,
+} from 'clients/api';
+
+interface IUseSupplyArgs {
+  asset: Asset;
+  account: string | undefined;
+}
+
+export type UseSupplyParams =
+  | Omit<ISupplyInput, 'tokenContract' | 'assetId' | 'account'>
+  | Omit<ISupplyBnbInput, 'tokenContract' | 'assetId' | 'account'>;
+
+const useSupply = ({ asset, account }: IUseSupplyArgs) => {
+  const { mutate: supplyNonBnb, isLoading: isSupplyNonBnBLoading } = useSupplyNonBnb({
+    assetId: asset?.id as VTokenId,
+    account,
+  });
+  const { mutate: supplyBnb, isLoading: isSupplyBnbLoading } = useSupplyBnb({ account });
+  if (asset.id === 'bnb') {
+    return { supply: supplyBnb, isLoading: isSupplyBnbLoading };
+  }
+  return { supply: supplyNonBnb, isLoading: isSupplyNonBnBLoading };
```

Comment:
The other props returned by the mutations are useful too, we should return them all:
```suggestion
const { supplyNonBnb, ...useSupplyNonBnbResult } = useSupplyNonBnb({
  assetId: asset.id as VTokenId,
  account,
});
const { supplyBnb, ...useSupplyBnbResult } = useSupplyBnb({ account });

if (asset.id === 'bnb') {
  return { supply: supplyBnb, ...useSupplyBnbResult };
}

return { supply: supplyNonBnb, ...useSupplyNonBnbResult };
```